SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -75,7 +59,46 @@ def _launch_makenv(self) -> subprocess.Popen:
 
         return subprocess.Popen([self.makenv_path, *arguments], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
 
-    async def _send_line(self, line: str):
+    async def _process_first_get(self, makenv_process) -> None:
+        """
+        Reads from the stdout of the makenv.sh process and sends it to the frontend.
+        This is done by the initial get request, which triggers the makenv.sh process.
+        """
+        SwanCustomEnvironmentsApiHandler.writing_log = True
+
+        # The first and any subsequent get requests will read from the process
+        # and stream out its output
+        with open(self.LOG_FILE, "w") as log_file:
+            for line in iter(makenv_process.stdout.readline, b""):
+                line = line.decode("utf-8")
+                log_file.write(line)
+                await self._send_line(line)
+
+        SwanCustomEnvironmentsApiHandler.writing_log = False
+
+    async def _process_subsequent_get(self, makenv_process) -> None:
+        """
+        Reads from a log file and sends to the frontend the output generated by the makenv.sh process.
+        This is done by any subsequent get request following the initial one, which triggers the makenv.sh process.
+        """
+        ret_code = makenv_process.poll()
+        if ret_code is None:
+            # The process is still running or has failed, 
+            # so we will read from the log file and send its content
+            with open(self.LOG_FILE, "r") as log_file:
+                while SwanCustomEnvironmentsApiHandler.writing_log:
+                    # If the log file is being written, we will read it and send its content
+                    line = log_file.readline()
+                    if line:
+                        await self._send_line(line)
+        elif ret_code != 0:
+            # The process has failed, so we will read the log file and send all its content
+            with open(self.LOG_FILE, "r") as log_file:
+                lines = "".join(log_file.readlines())
```

Comment:
Not super likely, but better to read and send line by line, in case the log file is very big? Just to be on the safe side.